### PR TITLE
Add eureka client

### PR DESCRIPTION
**Pull request by Bolo89**
- fix https://github.com/jhipster/jhipster-lite/issues/280

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Thanks @Bolo89 : I'll review it soon

**Comment by pascalgrimaud** [comment]
I think we should add some configs here: https://github.com/jhipster/jhipster-sample-app-microservice/blob/main/src/main/resources/config/application.yml#L27-L42

The current properties are not enough IMO

Currently:
image

The instance id should be like this:
image

File: src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/domain/EurekaDomainService.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  public void addProperties(Project project) {
+    project.addDefaultConfig(BASE_NAME);
+    projectRepository.template(project, getPath(SOURCE, "src"), BOOTSTRAP_PROPERTIES_FILE_NAME, getPath(MAIN_RESOURCES, CONFIG_FOLDER));
+    projectRepository.template(
+      project,
+      getPath(SOURCE, "src/test"),
+      BOOTSTRAP_PROPERTIES_FILE_NAME,
+      getPath(TEST_RESOURCES, CONFIG_FOLDER)
+    );
+  }
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/domain/EurekaDomainService.java:62]: I think this method should be smart, I mean:
- if the bootstrap.properties already exist, we should add the config in existing files
- if the file doesn't exist, we could create it like this

So I think the feature spring cloud config should be updated with the similar logic

File: src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/infrastructure/primary/EurekaResource.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class EurekaResource {
+
+  private final EurekaApplicationService eurekaApplicationService;
+
+  public EurekaResource(EurekaApplicationService eurekaApplicationService) {
+    this.eurekaApplicationService = eurekaApplicationService;
+  }
+
+  @Operation(summary = "Add Spring Cloud Eureka Client")
+  @ApiResponse(responseCode = "500", description = "An error occurred while adding Spring Cloud Eureka Client config")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/infrastructure/primary/EurekaResource.java:26]:
```suggestion
  @ApiResponse(responseCode = "500", description = "An error occurred while adding Spring Cloud Eureka Client")
```

File: src/test/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/infrastructure/primary/EurekaResourceIT.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  @Test
+  void shouldInit() throws Exception {
+    // Given
+    ProjectDTO projectDTO = TestUtils.readFileToObject("json/chips.json", ProjectDTO.class);
+    if (projectDTO == null) {
+      throw new GeneratorException("Error when reading file");
+    }
+    projectDTO.folder(FileUtils.tmpDirForTest());
+
+    Project project = ProjectDTO.toProject(projectDTO);
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/infrastructure/primary/EurekaResourceIT.java:47]:
```suggestion
    ProjectDTO projectDTO = readFileToObject("json/chips.json", ProjectDTO.class).folder(tmpDirForTest());
    Project project = ProjectDTO.toProject(projectDTO);
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
@Bolo89 : don't hesitate to ping me if it's finished and it needed a 2nd review

File: src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/domain/EurekaDomainService.java (hunk 1 above)
Comment (Bolo89) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/eureka/domain/EurekaDomainService.java]: done, for eureka, consul and spring cloud config

File: src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/common/domain/SpringCloudCommonDomainService.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.List;
+import tech.jhipster.lite.common.domain.FileUtils;
+import tech.jhipster.lite.error.domain.GeneratorException;
+import tech.jhipster.lite.generator.project.domain.Project;
+import tech.jhipster.lite.generator.project.domain.ProjectRepository;
+
+public class SpringCloudCommonDomainService implements SpringCloudCommonService {
+
+  public static final String COMMENT_PROPERTIES_PREFIX = "#";
+  public static final String KEY_PROPERTY_SEPARATOR = "=";
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/common/domain/SpringCloudCommonDomainService.java:15]: I think we can move it to Constants

**Review by pascalgrimaud — DISMISSED** [review]

**Comment by pascalgrimaud** [comment]
needed some manuel tests from my side and it can be merged

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
thanks a lot for the work here, @Bolo89 !

**Comment by Bolo89** [comment]
Bounty claimed at: https://opencollective.com/generator-jhipster/expenses/65313

**Pull request merged**